### PR TITLE
[LLM] support trainer for gpt and llama pre-training.

### PR DESCRIPTION
**Pull request by ZHUI**
### PR types

New features
### PR changes

Others
### Description

support trainer for gpt and llama pre-training.

### CODE REVIEW

File: paddlenlp/transformers/llama/modeling.py [hunk 1]
```diff
@@ -216,6 +217,11 @@ def _expand_mask(mask, dtype, tgt_length):
     return masked_fill(inverted_mask, inverted_mask.cast("bool"), float(finfo(dtype).min))
 
 
+def rms_norm_fused(x_in, w, eps):
+    fused_ln = try_import("fused_ln")
+    return fused_ln.fused_rms_norm(x_in, w, eps)[0]
```
Comment (wawltor) [inline paddlenlp/transformers/llama/modeling.py:222]: 现在fused_ln是通过外挂算子来实现的吧

File: model_zoo/gpt/run_pretrain_trainer.py [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+        GPTForPretraining,
+    ),
+    "llama": (
+        LlamaConfig,
+        LlamaForCausalLM,
+    ),
+}
+
+from dataset import GPTDataset, get_train_valid_test_split_
+from modeling_llama_pp import LlamaForCausalLMPipe
```
Comment (wawltor) [inline model_zoo/gpt/run_pretrain_trainer.py]: 对LLaMA的支持删除吧，后续可以通过分支维护来实现

File: model_zoo/gpt/run_pretrain_trainer.py [hunk 3]
```diff
@@ -0,0 +1,488 @@
+# Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
```
Comment (wawltor) [inline model_zoo/gpt/run_pretrain_trainer.py]: 2022->2023

File: paddlenlp/transformers/llama/modeling.py (hunk 1 above)
Comment (ZHUI) [reply paddlenlp/transformers/llama/modeling.py:222]: 对

**Review by wawltor — DISMISSED** [review]
LGTM

**Review by wawltor — APPROVED** [review]
LGTM

**Pull request merged**